numpyro/distributions/util.py

```diff
@@ -215,6 +215,20 @@ def multinomial(key, p, n, shape=()):
     return _multinomial(key, p, n, n_max, shape)
 
 
+@partial(jit, static_argnums=(2,))
+def _geometric(key, p, shape):
+    if shape is None:
+        shape = jnp.shape(p)
+    dtype = get_dtype(p)
+    tiny = jnp.finfo(dtype).eps
+    u = random.uniform(key, shape, dtype, tiny)
+    return jnp.floor(jnp.log(u) / jnp.log1p(-p))
```

Comment:
nit: I think it is better to move this implementation to `sample` method.

Reply:
But does that mean JIT will not be used? I actually follow the practice of other distributions such as Categorical to put the sampling method in util.py.

Reply:
Actually, if you sample from `Geometric`, `random.uniform` will be compiled for you. We often put complicated samplers to `util` to simplify codes in the main implementation. We will move categorical sampler to `CategoricalProbs.sample` when its performance is improved (currently, it is slow when the number of categories are large).

Reply:
Just a side note: when you move this sampler to GeometricLogits, you can replace `jnp.log1p(-p)` by `-soft_plus(logits)`